src/hooks/viewStateFacet/useGetSarcophagusDetails.ts

```diff
@@ -0,0 +1,34 @@
+import { useContractRead } from 'wagmi';
+import { ViewStateFacet__factory } from '@sarcophagus-org/sarcophagus-v2-contracts';
+import { useNetworkConfig } from 'lib/config';
+import { Sarcophagus } from 'types';
+import { useEffect, useState } from 'react';
+
+export function useGetSarcophagusDetails({ sarcoId }: { sarcoId: string | undefined }) {
```

Comment:
I'm running into an issue with this hook returning undefined for the sarcophagus on the detail page 🤔 I tried deleting the 
```
else {
        setSarcophagus(undefined);
      }
```
just to see what would happen and the result was that sometimes I'd get the data and sometimes I'd get undefined. 

<img width="632" alt="image" src="https://user-images.githubusercontent.com/8015883/201020713-8b87365b-ec09-4506-92ca-03f3d5e3ce96.png">

Reply:
Yes, this is going to need a little more investigation, I think I need to change the way the getSarcophagus call works. And I need to add error handling. So we can do those together later?